modules/ghlint-snakeyaml/src/main/kotlin/net/twisterrob/ghlint/yaml/Node.toPermissions.kt

```diff
@@ -0,0 +1,13 @@
+package net.twisterrob.ghlint.yaml
+
+import net.twisterrob.ghlint.model.Access
+import net.twisterrob.ghlint.model.Permission
+import org.snakeyaml.engine.v2.nodes.Node
+
+public fun Map<Node, Node>.toPermissions(): List<Permission> {
+	return this.map { (key, value) ->
+		Access.fromString(value.text)?.let {
+			Permission.fromString(key.text, it)
+		} ?: throw IllegalArgumentException("Invalid access: ${value.text}")
```

Comment:
model should handle any future permissions, if GH adds one, and it's added to the JSON-schema, older versions of ghlint shouldn't break.

Reply:
add explicit test for fake future permission, (it'll fail validation so the test needs to use `checkUnsafe`)